samples/drivers/adc/src/main.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2020 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <drivers/adc.h>
+
+#if !DT_NODE_EXISTS(DT_PATH(zephyr_user)) || \
+	!DT_NODE_HAS_PROP(DT_PATH(zephyr_user), io_channels)
+#error "No suitable devicetree overlay specified"
+#endif
+
+#define ADC_NUM_CHANNELS	DT_PROP_LEN(DT_PATH(zephyr_user), io_channels)
+
+#if ADC_NUM_CHANNELS > 2
+#error "Currently only 1 or 2 channels supported in this sample"
+#endif
+
+/* Check that the same ADC is used for first and second phandle */
+#if ADC_NUM_CHANNELS == 2 && \
+	DT_PHANDLE_BY_IDX(DT_PATH(zephyr_user), io_channels, 0) != \
+	DT_PHANDLE_BY_IDX(DT_PATH(zephyr_user), io_channels, 1)
```

Comment:
Hmm, this isn't doing what you want it to. DT_PHANDLE_BY_IDX returns a node identifier, which [is not a value](https://docs.zephyrproject.org/latest/guides/dts/api-usage.html#node-identifiers-are-not-values).

I think what is likely happening is that comparing the two node identifiers like this is silently converted to `0 != 0` in the preprocessor. (I bet you would get a warning with `-Wundef`.)

If you really want to keep this check, maybe you can compare the `device` structure addresses intead?

Reply:
Oops, good spot. I'm going to wrap it into `DT_REG_ADDR()` to make it comparable.